src/mjolnir/shortcutbuilder.cc

```diff
@@ -315,11 +294,30 @@ uint32_t ConnectEdges(GraphReader& reader,
                       GraphId& endnode,
                       uint32_t& opp_local_idx,
                       uint32_t& restrictions,
-                      float& average_density) {
+                      float& average_density,
+                      float& total_duration,
+                      float& total_truck_duration) {
   // Get the tile and directed edge.
   const GraphTile* tile = reader.GetGraphTile(startnode);
   const DirectedEdge* directededge = tile->directededge(edgeid);
 
+  // Add edge and turn duration for car
+  auto const nodeinfo = tile->node(startnode);
+  auto const turn_duration = OSRMCarTurnDuration(directededge, nodeinfo, opp_local_idx);
+  total_duration += turn_duration;
+  auto const speed = directededge->speed();
+  // TODO: check speed for zero
+  auto const edge_duration = directededge->length() / (speed * kKPHtoMetersPerSec);
+  total_duration += edge_duration;
+
+  // Add edge and turn duration for truck
+  // Currently use the same as for cars. TODO: implement for trucks
+  total_truck_duration += turn_duration;
+  auto const truck_speed = tile->GetSpeed(directededge, kNoFlowMask, kInvalidSecondsOfWeek, true);
+  // TODO: check speed for zero
+  auto const edge_duration_truck = directededge->length() / (truck_speed * kKPHtoMetersPerSec);
+  total_truck_duration += edge_duration_truck;
+
   // Copy the restrictions and opposing local index. Want to set the shortcut
   // edge's restrictions and opp_local_idx to the last directed edge in the chain
   opp_local_idx = directededge->opp_local_idx();
```

Comment:
i dont even know why we need to set this at all. why not use the value from the edge directly and remove this parameter from the function?

Reply:
I'm not sure I get your idea... When the cycle that does `ConnectEdges` finishes we know the last node but I don't think we know the last edge added to the shortcut in order to obtain `opp_local_idx` from it.
Also it is useful for me to know this previous edge idx to calculate the heading of the incoming edge. Alternatively I could certainly reverse `shape` and get "direction along polyline" but current approach looks cheaper to me

Reply:
nevermind, i see now that it does the first one by iteslf and moves to the next, sorry about that